paddle/fluid/framework/data_feed.cc

```diff
@@ -102,13 +110,14 @@ template <typename T>
 void PrivateQueueDataFeed<T>::ReadThread() {
   std::string filename;
   while (PickOneFile(&filename)) {
-    file_.open(filename.c_str());  // is_text_feed
-    PADDLE_ENFORCE(file_.good(), "Open file<%s> fail.", filename.c_str());
+    int err_no = 0;
+    fp_ = fs_open_read(filename, &err_no, pipe_command_);
+    __fsetlocking(&*fp_, FSETLOCKING_BYCALLER);
+    thread_local string::LineFileReader reader;
     T instance;
-    while (ParseOneInstance(&instance)) {
+    while (ParseOneInstanceFromPipe(&instance)) {
```

Comment:
I think it is not good way to decorate ParseOneInstance method with "FromPipe", for it narrow the data sources. E.g. if I need to create a data feeder parsing data from files or file mappings, they are not "from pipe". As the result, I need to override ReadThread and append methods like ParseOneInstanceFromMmap in "PrivateQueueDataFeed", such a base class level. Otherwise, if ParseOneInstance could be kept, I could simply override this method in a concrete sub-class

Reply:
I think you can just write a c++ pipe binary program to use mmap for training. and print out the instance with the same format DataFeed can recognize. We will provide a C++ framework similar to data_generator.py in next version so that users can load binary data into pipe program.

Reply:
Good comrade

Reply:
@@:)